src/Aspire.Hosting.Azure.Functions/AzureFunctionsProjectResourceExtensions.cs

```diff
@@ -93,6 +95,29 @@ public static class AzureFunctionsProjectResourceExtensions
             });
     }
 
+    /// <summary>
+    /// Configures the Azure Functions project resource to use the specified port as its HTTP endpoint.
+    /// This method queries the launch profile of the project to determine the port to
+    /// use based on the command line arguments configure in the launch profile,
+    /// </summary>
+    /// <param name="builder">The resource builder for the Azure Functions project resource.</param>
+    /// <returns>An <see cref="IResourceBuilder{AzureFunctionsProjectResource}"/> for the Azure Functions project resource with the endpoint configured.</returns>
+    private static IResourceBuilder<AzureFunctionsProjectResource> WithFunctionsHttpEndpoint(this IResourceBuilder<AzureFunctionsProjectResource> builder)
+    {
+        var launchProfile = builder.Resource.GetEffectiveLaunchProfile();
+        int? port = null;
+        if (launchProfile is not null)
+        {
+            var commandLineArgs = CommandLineArgsParser.Parse(launchProfile.LaunchProfile.CommandLineArgs ?? string.Empty);
+            if (commandLineArgs is { Count: > 0 })
+            {
+                var indexOfPort = commandLineArgs.IndexOf("--port");
+                port = indexOfPort > -1 ? int.Parse(commandLineArgs[indexOfPort + 1], CultureInfo.InvariantCulture) : null;
+            }
+        }
+        return builder.WithHttpEndpoint(port: port, targetPort: port, isProxied: port == null);
```

Comment:
Not understanding why you are setting the target port and proxied.

```suggestion
        return builder.WithHttpEndpoint(port: port);
```

Reply:
It looks like Azure Functions prioritizes the port defined in the launch settings over the one that it receives in the command line args when it's configured, so the target port that we configure in the `WithArgs` callback is disregarded.

I believe this is the right change we want to make if we want to allow Functions to set the port the underlying worker will listen on based on launchsettings and mapping the same port value out.

Reply:
You are right. Since it's listening already that makes sense.